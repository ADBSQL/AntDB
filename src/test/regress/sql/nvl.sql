set grammar to oracle;
select nvl(1.4,4.8) from dual;
select nvl(4,5) from dual;
select nvl(1.4,4) from dual;
select nvl(4,4.8) from dual;
select nvl(exp(2),4.8) from dual;
select nvl(exp(2),exp(3)) from dual;
select nvl('3',exp(4)) from dual;
select nvl('s',exp(4)) from dual;
select nvl(exp(4),'s') from dual;
select nvl(exp(4),'3') from dual;
select nvl('',4) from dual;
select nvl(null,4) from dual;
select nvl('','') from dual;
select nvl(4,'') from dual;
select nvl(null,null) from dual;
select nvl(null,'') from dual;
