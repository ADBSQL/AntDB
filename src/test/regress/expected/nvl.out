set grammar to oracle;
select nvl(1.4,4.8) from dual;
 NVL(1.4,4.8)  
---------------
           1.4
(1 row)

select nvl(4,5) from dual;
 NVL(4,5)  
-----------
         4
(1 row)

select nvl(1.4,4) from dual;
 NVL(1.4,4)  
-------------
         1.4
(1 row)

select nvl(4,4.8) from dual;
 NVL(4,4.8)  
-------------
           4
(1 row)

select nvl(exp(2),4.8) from dual;
  NVL(EXP(2),4.8)   
--------------------
 7.3890560989306502
(1 row)

select nvl(exp(2),exp(3)) from dual;
 NVL(EXP(2),EXP(3))  
---------------------
  7.3890560989306502
(1 row)

select nvl('3',exp(4)) from dual;
 NVL('3',EXP(4))  
------------------
 3
(1 row)

select nvl('s',exp(4)) from dual;
 NVL('S',EXP(4))  
------------------
 s
(1 row)

select nvl(exp(4),'s') from dual;
ERROR:  invalid input syntax for type numeric: "s"
select nvl(exp(4),'3') from dual;
  NVL(EXP(4),'3')   
--------------------
 54.598150033144239
(1 row)

select nvl('',4) from dual;
 NVL('',4)  
------------
 4
(1 row)

select nvl(null,4) from dual;
 NVL(NULL,4)  
--------------
 4
(1 row)

select nvl('','') from dual;
 NVL('','')  
-------------
 
(1 row)

select nvl(4,'') from dual;
 NVL(4,'')  
------------
 4
(1 row)

select nvl(null,null) from dual;
 NVL(NULL,NULL)  
-----------------
 
(1 row)

select nvl(null,'') from dual;
 NVL(NULL,'')  
---------------
 
(1 row)

